src/message_dao.py

```diff
@@ -109,6 +109,15 @@ def get(self, message_id: str) -> Optional[Message]:
         msg.children = self._get_children(message_id)
         return msg
 
+    @lru_cache()
+    def previous_version_candidates(self, message: Message) -> List[Message]:
```

Comment:
bigger question: we've made PatchAssociator be an ABC, but the core association logic is here, in the MessageDao.

I'm not sure we can meaningfully make new implementations of the PatchAssociator on its own.

I don't know how we want to handle this, since we don't want to expose SQL to the associator, necessarily.

We could make this method be like `find_with_matching_subject(self, reduced_subject: str, from_: str)`, which is something we can implement in the fake dao...

Reply:
We could make another class called a PatchCandidateSelector (or something like that) which could be passed into the MessageDao from the PatchAssociator. The PatchCandidateSelector could effectively implement this method and would be tied to a PatchAssociator. Thoughts?

Reply:
Incorporated the `find_with_matching_subject` idea but made it more general (as discussed in 1:1 w/ Daniel). Now allows for expansion of matching to other attributes with quite ease.